Essentials/src/main/java/com/earth2me/essentials/commands/Commandhelpop.java

```diff
@@ -20,25 +23,37 @@ public Commandhelpop() {
     @Override
     public void run(final Server server, final User user, final String commandLabel, final String[] args) throws Exception {
         user.setDisplayNick();
-        final String message = sendMessage(server, user.getDisplayName(), args);
+        final String message = sendMessage(server, user, args);
         if (!user.isAuthorized("essentials.helpop.receive")) {
             user.sendMessage(message);
         }
     }
 
     @Override
     public void run(final Server server, final CommandSource sender, final String commandLabel, final String[] args) throws Exception {
-        sendMessage(server, Console.DISPLAY_NAME, args);
+        sendMessage(server, Console.getInstance(), args);
     }
 
-    private String sendMessage(final Server server, final String from, final String[] args) throws Exception {
+    private String sendMessage(final Server server, final IMessageRecipient from, final String[] args) throws Exception {
         if (args.length < 1) {
             throw new NotEnoughArgumentsException();
         }
-        final String message = tl("helpOp", from, FormatUtil.stripFormat(getFinalArg(args, 0)));
-        ess.getLogger().log(Level.INFO, message);
-        ess.broadcastMessage("essentials.helpop.receive", message);
-        return message;
+        final String message = FormatUtil.stripFormat(getFinalArg(args, 0));
+        final String finalMessage = tl("helpOp", from.getDisplayName(), message);
+        ess.getLogger().log(Level.INFO, finalMessage);
+        ess.broadcastMessage("essentials.helpop.receive", finalMessage);
+
+        final HelpopMessageSentEvent sentEvent = new HelpopMessageSentEvent(from, getOnlineHelpopRecipients(), message);
+        ess.getServer().getPluginManager().callEvent(sentEvent);
+
+        return finalMessage;
+    }
+
+    private List<? extends IMessageRecipient> getOnlineHelpopRecipients() {
+        return ess.getOnlinePlayers().stream()
+                .filter(player -> player.hasPermission("essentials.helpop.receive"))
+                .map(ess::getUser)
+                .collect(Collectors.toList());
```

Comment:
don't use streams and this list isn't mutable. changes to the list should effect who is sent the message.